plugins/com.google.cloud.tools.eclipse.login.test/src/com/google/cloud/tools/eclipse/login/ui/LabelImageLoaderTest.java

```diff
@@ -0,0 +1,123 @@
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import com.google.cloud.tools.eclipse.test.util.ui.ShellTestResource;
+import java.net.MalformedURLException;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.graphics.ImageData;
+import org.eclipse.swt.graphics.PaletteData;
+import org.eclipse.swt.graphics.RGB;
+import org.eclipse.swt.widgets.Label;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class LabelImageLoaderTest {
+
+  static final byte[] someImageBytes = { 71, 73, 70, 56, 57, 97, 1, 0, 1, 0, 0, 0, 0, 33,
+      (byte) 249, 4, 1, 0, 0, 0, 0, 44, 0, 0, 0, 0, 1, 0, 1, 0, 0, 2 };
+  static final ImageData someImageData = new ImageData(
+      1, 1, 1, new PaletteData(new RGB[] { new RGB(0, 0, 0) }));
+
+  @Rule public ShellTestResource shellResource = new ShellTestResource();
+
+  private final LabelImageLoader imageLoader = new LabelImageLoader();
+  private Label label;
+
+  @Before
+  public void setUp() {
+    label = new Label(shellResource.getShell(), SWT.NONE);
+  }
+
+  @After
+  public void tearDown() {
+    Image image = label.getImage();
+    label.dispose();
+    if (image != null) {
+      assertTrue(image.isDisposed());
+    }
+
+    LabelImageLoader.cache.clear();
+  }
+
+  @Test
+  public void testLoadImage_nullImageUrl() throws MalformedURLException {
+    try {
+      imageLoader.loadImage(null, label, 1, 1);
+      fail();
+    } catch (NullPointerException ex) {}
+  }
+
+  @Test
+  public void testLoadImage_malformedImageUrl() {
+    try {
+      imageLoader.loadImage("marlformed", label, 1, 1);
+    } catch (MalformedURLException ex) {}
+  }
+
+  @Test
+  public void testConstructor_nonPositiveWidth() throws MalformedURLException {
+    try {
+      imageLoader.loadImage("http://example.com", label, 0 /* zero width */, 10);
+      fail();
+    } catch (IllegalArgumentException ex) {}
+
+    try {
+      imageLoader.loadImage("http://example.com", label, -1 /* negative width */, 10);
+      fail();
+    } catch (IllegalArgumentException ex) {}
+  }
+
+  @Test
+  public void testConstructor_nonPositiveHeight() throws MalformedURLException {
+    try {
+      imageLoader.loadImage("http://example.com", label, 10, 0 /* zero height */);
+      fail();
+    } catch (IllegalArgumentException ex) {}
+
+    try {
+      imageLoader.loadImage("http://example.com", label, 10, -1 /* negative height */);
+      fail();
+    } catch (IllegalArgumentException ex) {}
+  }
+
+  @Test
+  public void testStoreInCache() {
+    assertTrue(LabelImageLoader.cache.isEmpty());
+    LabelImageLoader.storeInCache("http://example.com", someImageData);
+    assertEquals(1, LabelImageLoader.cache.size());
+    assertTrue(someImageData == LabelImageLoader.cache.get("http://example.com"));
```

Comment:
Since I am testing caching, I am explicitly testing if the cached instance is the given instance. I'll attach a message to the assertion to clarify this.